lang/ruby/lib/avro/io.rb

```diff
@@ -397,8 +397,11 @@ def read_default_value(field_schema, default_value)
           return Integer(default_value)
         when :float, :double
           return Float(default_value)
-        when :boolean, :enum, :fixed, :string, :bytes
+        when :boolean, :enum, :fixed, :string
           return default_value
+        when :bytes
+          return default_value unless field_schema.type_adapter
+          return field_schema.type_adapter.decode(default_value)
```

Comment:
[for reviewer] or shall I only use `type_adapter` for decimal logical type? Is there a case where `type_adapter` exists, but `default_value` is already decoded in schema definition?

Reply:
as my test only cover Decimal, I think it is safe to just use `type_adapter` for Decimal only. Separate PR would need if same fix for other logical type

Reply:
[53e2f59](https://github.com/apache/avro/pull/2277/commits/53e2f59c33aff9b9ae59bbf83653c9fa627facb2)